lib/haml_lint/tree/tag_node.rb

```diff
@@ -20,6 +20,15 @@ def contains_script?
       @value[:parse] && !@value[:value].strip.empty?
     end
 
+    # Returns whether this tag has a specified attribute
+    # `=`).
+    #
+    # @return [true,false]
+    def has_hash_attribute?(attribute)
+      hash_attributes? &&
+        hash_attributes_source.scan(/(#{attribute})\W/).any?
```

Comment:
@sds the parser gem does not give access to the hash keys as nodes, so I went with a regex. I'll refine this matcher, but is it an ok direction from your standpoint?